src/components/auth/Login.jsx

```diff
@@ -1,14 +1,93 @@
+import { AuthProvider, useAuth } from "../../AuthProvider";
+import { Link, useNavigate } from "react-router-dom";
+import { useEffect, useState } from "react";
+
 import FormInput from "../common/FormInput";
+import { login } from "../../util/fetchData"
+
 const Login = () => {
+  const [formData, setFormData] = useState({ email: "", password: "" });
+  const [formErrors, setFormErrors] = useState({});
+  const { isLoggedIn, loginAction } = useAuth();
+  const navigate = useNavigate();
+
+  const handleChange = (e) => {
+    setFormData({ ...formData, [e.target.name]: e.target.value });
+    setFormErrors({ ...formErrors, [e.target.name]: '', form: '' });
+  }
+
+  const handleSubmit = async (e) => {
```

Comment:
This component is doing a lot. There is form validation and user login. Best to split both into their own component